RxSwift/Schedulers/VirtualTimeScheduler.swift

```diff
@@ -195,15 +195,29 @@ open class VirtualTimeScheduler<Converter: VirtualTimeConverterType>
         self.running = false
     }
 
-    /// Advances the scheduler's clock by the specified relative time.
-    public func sleep(_ virtualInterval: VirtualTimeInterval) {
+    /// Advances the scheduler's clock by the specified relative time interval.
+    ///
+    /// When `deep` is `true` it also advances time of all already scheduled item by the interval.
+    /// This allows to emulate computer sleep (suspend) during which all timers are "frozen".
+    ///
+    public func sleep(_ virtualInterval: VirtualTimeInterval, deep: Bool = false) {
```

Comment:
I thought of this some more and I think I can "live with it" since you presented a solid use-case.
I don't like the naming `deep` though - I think something that indicates it pushes all events simultaneously/synchronized/interlocked makes more sense.

Either:

```suggestion
    public func sleep(_ virtualInterval: VirtualTimeInterval, synchronized: Bool = false) {
```

O:

```suggestion
    public func sleep(_ virtualInterval: VirtualTimeInterval, interlocked: Bool = false) {
```

WDYT ?

Reply:
👍 I'l go with `interlocked`.